Sources/Baggage/BaggageContext.swift

```diff
@@ -32,16 +32,25 @@
 ///       }
 ///     }
 public struct BaggageContext {
-    private var _storage = [ObjectIdentifier: ValueContainer]()
+    private var _storage = [AnyBaggageContextKey: ValueContainer]()
 
     /// Create an empty `BaggageContext`.
     public init() {}
 
     public subscript<Key: BaggageContextKey>(_ key: Key.Type) -> Key.Value? {
         get {
-            self._storage[ObjectIdentifier(key)]?.forceUnwrap(key)
+            self._storage[AnyBaggageContextKey(key)]?.forceUnwrap(key)
         } set {
-            self._storage[ObjectIdentifier(key)] = newValue == nil ? nil : ValueContainer(value: newValue!)
+            self._storage[AnyBaggageContextKey(key)] = newValue.map {
+                ValueContainer(value: $0)
+            }
+        }
+    }
+
+    public var baggageItems: [AnyBaggageContextKey: Any] {
+        // TODO: key may not be unique
+        self._storage.reduce(into: [:]) {
+            $0[$1.key] = $1.value.value
         }
```

Comment:
I'd kind of hope we would not need to allocate a new dictionary, but we can solve this in a separate PR; we can also add some performance benchmarks and then solve that; ticket perhaps for this?

Reply:
Yes, will be included in the "tradeoffs" ticket after the merge.